probe-rs/src/probe/daplink/tools.rs

```diff
@@ -180,7 +180,11 @@ pub fn open_device_from_selector(
             };
 
             let timeout = Duration::from_millis(100);
-            let language = handle.read_languages(timeout)?[0];
+            let language = match handle.read_languages(timeout)?.get(0) {
+                Some(lang) => lang.clone(),
+                None => continue,
+            };
+
             let sn_str = handle
```

Comment:
So this `sn_str` variable actually needs to become None if `language` is None too. That should do the trick :)

Reply:
So, as I did in the commit I managed to slip in just a few seconds before your comment? :D

Reply:
haha great timing :D